profiles/events.py

```diff
@@ -0,0 +1,33 @@
+import logging
+from typing import Callable
+
+from elife_bus_sdk.events import ProfileEvent
+from elife_bus_sdk.publishers import EventPublisher
+from flask import Flask
+
+from profiles.exceptions import UpdateEventFailure
+from profiles.models import Affiliation, EmailAddress, Profile
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def send_update_events(publisher: EventPublisher) -> Callable[..., None]:
+    def wrapper(sender: Flask, changes: tuple) -> None:  # pylint:disable=unused-argument
+        ids = []
+
+        for instance, operation in changes:  # pylint:disable=unused-variable
+            if isinstance(instance, Profile):
+                ids.append(instance.id)
+            if isinstance(instance, (Affiliation, EmailAddress)):
```

Comment:
It's a bit worrying that we would have to traverse the graphs from any modified object to go back to what is the `profile` being modified, but it's good for now :+1: 
It comes from the fact that the ORM is fine-grained for the single entities, but we are after a model where `1 transaction` == `1 Profile Aggregate` == `1 Profile object and any of the Value Object it contains`

Reply:
I agree. If it was not for composite types being used we could just look for `Profile` model changes as any direct change to any of its Foreign keys is picked up, its just that the Foreign key's composite types do not get classed as a `Profile` model update so you have to account of the other `db.models`.

But yes its fine for now.

Reply:
Composites are the problem? I'm not entirely happy with how they work, so we could just replace them with plain value objects. (Though this actually sounds like an upstream bug.)

Reply:
They are not a problem, they are just the current reason for having to check per model type. 

I am used to them being plain value objects, but there will still be an element of checking based on model types even if you do that. 

If say `Address` was a `db.Model` and had a foreign key relationship to an `Affiliation` and you updated the `Address` object, you would still only get updates up to the `Affiliation` (would not bubble up to a `Profile` update), that's why we have to pull the profile id from the `Affiliation` object in that case.

For this use case I think its fine, its only run on a `session.commit` and will only send one event message per profile update.

Reply:
Ah ok, I misunderstood.